### PR TITLE
add SubjectAltNames to tls.validatecontext for sdsconfig

### PR DESCRIPTION
**Pull request by quanjielin**
https://github.com/istio/istio/issues/6411

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 1]
```diff
@@ -516,10 +517,15 @@ func applyUpstreamTLSSettings(cluster *v2.Cluster, tls *networking.TLSSettings,
 				},
 			}
 		} else {
-			cluster.TlsContext.CommonTlsContext.ValidationContextType = model.ConstructValidationContext(model.CARootCertPath)
+			cluster.TlsContext.CommonTlsContext.ValidationContextType = model.ConstructValidationContext(model.CARootCertPath, tls.SubjectAltNames)
 			cluster.TlsContext.CommonTlsContext.TlsCertificateSdsSecretConfigs = []*auth.SdsSecretConfig{}
 			refreshDuration, _ := ptypes.Duration(meshConfig.SdsRefreshDelay)
 			for _, sa := range serviceAccounts {
+				// Skip service account which is added from annotation 'alpha.istio.io/canonical-serviceaccounts'
```
Comment (wattli) [inline pilot/pkg/networking/core/v1alpha3/cluster.go]: I think a better comment would be 'why we want to something' instead of 'what it does'  :)

Comment (lei-tang) [reply pilot/pkg/networking/core/v1alpha3/cluster.go]: +1, other than that, the PR LGTM.

Comment (quanjielin) [reply pilot/pkg/networking/core/v1alpha3/cluster.go]: added more comments to explain the reason.

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 2]
```diff
@@ -516,10 +517,16 @@ func applyUpstreamTLSSettings(cluster *v2.Cluster, tls *networking.TLSSettings,
 				},
 			}
 		} else {
-			cluster.TlsContext.CommonTlsContext.ValidationContextType = model.ConstructValidationContext(model.CARootCertPath)
+			cluster.TlsContext.CommonTlsContext.ValidationContextType = model.ConstructValidationContext(model.CARootCertPath, tls.SubjectAltNames)
 			cluster.TlsContext.CommonTlsContext.TlsCertificateSdsSecretConfigs = []*auth.SdsSecretConfig{}
 			refreshDuration, _ := ptypes.Duration(meshConfig.SdsRefreshDelay)
 			for _, sa := range serviceAccounts {
+				// Skip service account which is added from annotation 'alpha.istio.io/canonical-serviceaccounts' to avoid fetching duplicated
+				// cert from CA, the service account added in annotation in only used for building VerifySubjectAltName in TLS ValidationContext.
```
[the remark this answers is not in the document]
Comment (quanjielin) [reply pilot/pkg/networking/core/v1alpha3/cluster.go]: updated

**Pull request merged**